### DISCUSSION TITLE
Framework UI para React?

### DISCUSSION THREAD

**Discussion by JosielLima** [frontendbr/forum#1014] [Perguntas e Respostas]
Olá galera, pode parecer uma pergunta repetida, mas não está simples definir um framework CSS (uso obrigatório) para os próximos projetos da empresa em React. O time tem preferência por frameworks que possuem base no material design. Ao pesquisar surgiram algumas dúvidas?

1) Como vocês trabalham o CSS no React? Criam do zero? Usam lib própria? Reactstrap?

2) Já utilizaram o material-ui-next de react? Ele parece engessado, pois gera CSS inline, o quanto é difícil personalizar os componentes?

3) É possível usar o Materialize ou bootstrap material e ignorar os arquivos de Jquery?

**Comment by ninetails** [comment]
Disclaimer: opinião própria, de acordo com a minha vivência.

1. Geralmente do zero, pois realmente é difícil achar uma lib onde o seu build está alinhado com o build já feito. Claro que o ideal é fazer um projeto só pros componentes, com um storybook e bem testado.
2. Já utilizei o material-ui, achei muito bom realmente pelo fato de ser CSS-in-JS e não te adicionar nenhum build externo: stylus, sass, less etc que às vezes nem é o que eu uso no projeto. Se nada te atender, recomendo procurar por Componentes ao invés de libs completas, mas lib de UI tb tem várias e também feito por vários vendors bem conhecidos
3. jQuery + React? A idéia de tentar usar React é tentar ~fugir de~ evitar implementações que usem js imperativo (no mundo ideal, claro.. fatalmente um dia c vai precisar usar ref, e com ele daria para utilizar jQuery talvez, mas nunca é recomendável)

Sempre lembrando que não necessariamente pq foi decidido usar React de stack que será mais ágil para desenvolver, ainda mais no começo.

**Comment by doug2k1** [comment]
Estou em um projeto que é muito baseado em Material Design e estamos usando [React MD](https://react-md.mlaursen.com). Atende bem e é bem customizável através de variáveis do Sass. Tem um ou outro bug que temos que contornar, mas é mais produtivo do que se fôssemos criar tudo do zero.

**Comment by leticiacostadias** [comment]
Por partes:

1. Nos projetos que trabalhei, quando criamos os estilos do zero, usamos sass separando estilos base em outro diretório e, na pasta do componente, criamos um arquivo para importar no jsx do componente (se ele precisar de css específico).
2. Em outros projetos já utilizei o [AntD](https://ant.design/), o [Semantic UI](https://react.semantic-ui.com) e o [react-bootstrap](https://react-bootstrap.github.io/). Todos bem simples de utilizar e já resolvem bem, na maioria das vezes.
3. Como comentei que já usei, tem o rebuilt do Bootstrap para react e, procurando rapidamente no Google, achei o rebuilt do [Materialize](https://react-materialize.github.io) também

Usando esses frameworks, tento evitar fazer edição do css deles, até pra não ter muita dor de cabeça.  No caso do AntD, eles usam bastante estilos inline nos exemplos da documentação, porém, eu prefiro adicionar alguns arquivos meus pra fazer isso de maneira mais limpa.

**Comment by leticiacostadias** [comment]
Hoje o Medium ainda trouxe [mais algumas opções](https://hackernoon.com/23-best-react-ui-component-libraries-and-frameworks-250a81b2ac42)
